File: reac-typescript/booking-client/src/models/Property.ts
import { BaseModel } from "./BaseModel";
export interface Property extends BaseModel {
  Id: string;
  Name: string;
  PropertyUrl: string;
  CityUrl: string;
  CityId: string | null;
  CityName: string;
  Street: string;
  District: string;
  AvgPrice: number;
  Rule: string;
  Description: string;
  UserId: string;
  UserMail: string;
  Services: string[] | null;
  ImageProperty: string | null;
  Rooms: string[] | null;
  Tags: string[] | null;
}

File: BackEnd/BookingSystem/BookingSystem.Domain/Entity/Property.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BookingSystem.Domain.Entity
{
    public class Property : BaseEntity<string>
    {
        [StringLength(150)]
        public string Name { get; set; } = string.Empty;
        [StringLength(150)]
        public string PropertyUrl { get; set; } = string.Empty;
        public int CityId { get; set; }
        public string CityUrl { get; set; } = string.Empty;
        [StringLength(150)]
        public string Street { get; set; } = string.Empty;
        [StringLength(150)]
        public string District { get; set; } = string.Empty;
        public decimal AvgPrice { get; set; }
        public string Rule { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public string UserId { get; set; } = string.Empty;
        public ApplicationUser User { get; set; }
        public City City { get; set; }
        public ICollection<Service> Services { get; set; }
        public ICollection<ImageProperty> ImageProperties { get; set; }
        public ICollection<Room> Rooms { get; set; }
        public ICollection<Tag> Tags { get; set; }
    }
}
